dsl/camel-kamelet-main/src/main/java/org/apache/camel/main/KameletMain.java

```diff
@@ -732,6 +734,21 @@ protected CamelContext createCamelContext() {
             throw RuntimeCamelException.wrapRuntimeException(e);
         }
 
+        // Start the context service loader plugin after all dependencies and downloaders are set up
+        // This ensures the classpath is enhanced and ServiceLoader can discover third-party plugins
+        ContextServiceLoaderPluginResolver contextServicePlugin = answer
+                .getCamelContextExtension().getContextPlugin(ContextServiceLoaderPluginResolver.class);
+        if (contextServicePlugin != null) {
+            try {
+                // force start context service loader plugin to discover and load third-party plugins
+                ServiceHelper.startService(contextServicePlugin);
+            } catch (Exception e) {
+                LOG.warn(
```

Comment:
Again I think this exception should not be caught but to let it fail fast instead.